lib/spack/docs/build_systems/intelpackage.rst

```diff
@@ -0,0 +1,607 @@
+
+  * To proceed with an externally configured license, leave the newly templated
+    file as is (containing comments only) and close the editor. You do not need
+    to touch the file again.
+
+  * To configure your own license, copy the contents of your downloaded license
+    file into the opened file, save it, and close the editor.
+
+  To revisit and manually edit this file, such as prior to a subsequent
+  installation attempt, find it at
+  ``$SPACK_ROOT/etc/spack/licenses/intel/intel.lic`` .
+
+  Spack will place symbolic links to this file in each directory where licensed
+  Intel binaries were installed.  If you kept the template unchanged, Intel tools
+  will simply ignore it.
+
+**TODO:** `PR #6534 "Intel v18 License File Format Issue"
+<https://github.com/spack/spack/issues/6534>`_.
+
+----------------------------------------------------------
+Integration of Intel tools installed *external* to Spack
+----------------------------------------------------------
+
+This section discusses `route 1`_ from the introduction.
+
+A site that already uses Intel tools, especially licensed ones, will likely
+have some versions already installed on the system, especially at a time when
+Spack is just being introduced. It will be useful to make such previously
+installed tools available for use by Spack as they are. How to do this varies
+depending on the type of the tools:
+
+^^^^^^^^^^^^^^^^^^^^^^
+Integrating compilers
+^^^^^^^^^^^^^^^^^^^^^^
+
+Configure external Intel compilers, like all compilers that Spack is to use,
+in ``compilers.yaml`` files located in
+``$SPACK_ROOT/etc/spack/`` or your own ``~/.spack/`` directory.
+In the Spack documentation, see
+`Configuration Files in Spack <http://spack.readthedocs.io/en/latest/configuration.html>`_
+in general and
+`Vendor-Specific Compiler Configuration <http://spack.readthedocs.io/en/latest/getting_started.html#vendor-specific-compiler-configuration>`_,
+section Intel Compilers.
+
+Briefly, the ``compilers.yaml`` files combine C and Fortran compilers of a
+specific vendor release and define such a set as a Spack
+`spec <http://spack.readthedocs.io/en/latest/basic_usage.html#specs-dependencies>`_
+that in this case has the form ``intel@compilerversion`` [fn7]_.
+The entry determines how the spec is to be resolved, via ``paths`` and/or
+``modules`` tokens, to each language compiler in the set.
+
+The following example illustrates how to integrate the 2018 Intel compiler
+suite, which outside of Spack was activated by users of the example system as
+``module load intel/18``. Since Spack must be rather more picky about versions,
+we must specify full paths and complete modulefile names in the relevant
+``compilers.yaml`` entry:
+
+.. code-block:: yaml
+
+    compilers:
+    - compiler:
+        spec:       intel@18.0.2
+        operating_system:   centos6
+        target:     x86_64
+        modules:    [intel/18/18.0.2]
```

Comment:
shall one also load certain `gcc` module?

Reply:
Good point to be explicit about that. I'll add:

> The Intel compilers need and use GCC to provide certain functionality, notably
to support C++. In the preceding minimal example, the  system's default ``gcc``
command would be queried for such needs.  To customize the GCC integration, see
`Vendor-Specific Compiler Configuration
<http://spack.readthedocs.io/en/latest/getting_started.html#vendor-specific-compiler-configuration>`_ 
in the Spack documentation.  In addition to the ways shown there, the 
modulefile that provides the Intel compilers (``intel/18/18.0.2`` in the 
example) might, for the benefit of users outside of Spack, already specify
another ``gcc`` version via compiler flag environment variables or (hopefully
not) via a sneaky extra ``PATH`` addition.